### PR TITLE
Make empty functions follow standard AbstractSet interface:

### PR DESCRIPTION
**Pull request by NHDaly**
- Make `empty!()` return the modified set
- Implement `empty(s::SortedSet, t::Type)`
- Implement `Base.emptymutable(s::SortedSet, t::Type)` (alias for above)

This is a follow-up to #571, and implements some of the missing parts of the AbstractSets interface.

### CODE REVIEW

File: src/sorted_set.jl
Comment (oxinabox) [inline src/sorted_set.jl]: Can we flip these definitions:

so that `empty` called `emptymutable`?
c.f. https://github.com/JuliaLang/julia/pull/34676

Or if we want to follow current practice of base then both should just directly construct the `SortedSet`

Comment (NHDaly) [reply src/sorted_set.jl:575]: 👍 yeah that makes sense to me. Sure.

Comment (NHDaly) [reply src/sorted_set.jl:575]: Done!

**Comment by NHDaly** [comment]
> Can we add a few tests?
> Should be trivial.

Sorry this took so long. :) Thanks, good call. Done!

**Comment by NHDaly** [comment]
PTAL: This is ready for review.

**Review by oxinabox — APPROVED** [review]

**Review by oxinabox — APPROVED** [review]
I have made you a maintainer.
Follow [ColPrac](https://github.com/SciML/ColPrac) it has everything i would normally tell you written down.

This PR looks good to me.
It needs rebasing and lets hold off merging it til @eulerkochy finishes GSOC stuff,
then lets change the version to 0.19-DEV.

Arguably this is not breaking since it was a bug before where disagreed with Base.
But since we are planning on swapping to 0.19-DEV soon anyway, I am inclined to just include this then.

**Comment by NHDaly** [comment]
> I have made you a maintainer.
> Follow ColPrac it has everything i would normally tell you written down.

Thanks Lyndon! That's a great idea. I'm sorry it took me so long to respond; i've been super busy both with work and also the local elections here! :)

😱 it looks like the invitation can expire, and i've let it do so? 😱 Sorry! If you still think it's useful, please do add me again and I'll accept it right away this time!
~Nathan

**Comment by oxinabox** [comment]
re-sent

**Comment by NHDaly** [comment]
> It needs rebasing and lets hold off merging it til @eulerkochy finishes GSOC stuff,
> then lets change the version to 0.19-DEV.

Thanks! I've bumped the version to 0.19.0-DEV. (Is that right? Or did you literally mean `0.19-DEV`?)

Is the GSOC stuff finished? Let me know and i can merge :)

**Comment by eulerkochy** [comment]
@NHDaly yes, you can go ahead and merge. #651 is still unfinished, but I don't think I'll be able to touch that PR for quite sometime now (hectic schedule of classes 😢 ).

**Comment by NHDaly** [comment]
😢 no worries! <3 very much understood. I hope classes are going well! That's of course the most important thing :)

okay, i'll merge this now then! :)

**Pull request merged**
